src/view_maybe/view_maybe.h

```diff
@@ -140,27 +101,22 @@ class ref_maybe_view
 };
 
 namespace view {
-struct __maybe_fn {
-    template <Nullable T>
-    constexpr auto operator()(T&& t) const
-        noexcept(noexcept(ref_maybe_view{std::forward<T>(t)}))
-        requires std::is_reference_v<T> &&
-        requires {ref_maybe_view{std::forward<T>(t)};}
-    {
-        return ref_maybe_view{std::forward<T>(t)};
-    }
-
-    template <Nullable T>
-    constexpr auto operator()(T&& t) const
-        noexcept(noexcept(safe_maybe_view{std::forward<T>(t)}))
-        requires !std::is_reference_v<T> &&
-        requires {safe_maybe_view{std::forward<T>(t)};}
-    {
-        return safe_maybe_view{std::forward<T>(t)};
-    }
-};
-
-inline constexpr __maybe_fn maybe{};
+    struct __maybe_fn {
+        template <Nullable T>
+        constexpr ref_maybe_view<T> operator()(T& t) const noexcept {
+            return ref_maybe_view<T>{t};
+        }
+
+        template <class T, Nullable U = std::remove_cv_t<T>>
+            requires ranges::Constructible<U, T>
```

Comment:
Ditto propagate `CopyConstructible` requirement:
```suggestion
            requires ranges::Constructible<U, T> && ranges::CopyConstructible<U>
```